tendrl/gluster_integration/manager/manager.py

```diff
@@ -200,6 +266,27 @@ def on_pull(self, raw_data, cluster_id):
                     index += 1
                 except KeyError:
                     break
```

Comment:
@nthomas-redhat the point is in valid case as well when all the volumes exhaust we depend on this KeyError to break the infinite while loop. If we change it to continue, it would never break. As we dont know the actula no of volume beforehand, I am not sure we can use a for loop to achieve this. We might have to retain the logic as is I feel. Chances of KeyError are not possible as data comes from a gluster generated file, so it not a risky deal I feel :)